File: src/main/java/ru/dargen/evoplus/event/inventory/InventorySlotUpdateEvent.java
package ru.dargen.evoplus.event.inventory;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import net.minecraft.item.ItemStack;
import ru.dargen.evoplus.event.CancellableEvent;

@Getter @Setter
@AllArgsConstructor
public class InventorySlotUpdateEvent extends CancellableEvent {

    protected int syncId;
    protected int slot;
    protected ItemStack stack;
    protected InventoryOpenEvent openEvent;
    protected boolean hidden;

}

File: src/main/kotlin/ru/dargen/evoplus/api/event/inventory/InventorySlotUpdateEvent.kt
package ru.dargen.evoplus.api.event.inventory

import net.minecraft.item.ItemStack

class InventorySlotUpdateEvent(
    syncId: Int,
    var slot: Int,
    var stack: ItemStack,
    var openEvent: InventoryOpenEvent?,
    var isHidden: Boolean
) : InventoryEvent(syncId)